### ISSUE TITLE
Cluster-wide BackupPolicies

### ISSUE THREAD

**Issue by NaPs** [cybozu-go/moco#437]
In its current state, as cluster administrators we are unable to provide ready-to-use BackupPolicies to our users.

It could be nice to either have a cluster-wide BackupPolicy resource (ClusterBackupPolicy?) or allow a MysqlCluster to reference a BackupPolicy located in another namespace. In both ways, there will be a challenge managing the secret holding the credentials to the S3 storage and an ideal solution will implies not running Cronjobs into the same namespace as MysqlCluster to prevent users from inspecting CronJob's Pods and get S3 credentials.

**Comment by NaPs** [comment]
As a workaround, here is our solution using a Kyverno policy to replicate the BackupPolicy and secret in any namespace. This is not perfect since it force us to share S3 credentials with users.

```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: sync-moco-backup
spec:
  rules:
  - name: sync-daily-backuppolicy
    match:
      resources:
        kinds:
        - Namespace
    generate:
      apiVersion: moco.cybozu.com/v1beta2
      kind: BackupPolicy
      name: daily
      namespace: "{{request.object.metadata.name}}"
      synchronize: true
      clone:
        namespace: moco-system
        name: daily
  - name: sync-backup-secret
    match:
      resources:
        kinds:
        - Namespace
    generate:
      apiVersion: v1
      kind: Secret
      name: moco-backup-secret
      namespace: "{{request.object.metadata.name}}"
      synchronize: true
      clone:
        namespace: moco-system
        name: moco-backup-secret
```

**Comment by masa213f** [comment]
Thank you for the proposal. It sounds good for some use cases.
I discussed this proposal with my team and concluded it was hard to implement now.

In the current design, we assumed a user who creates MySQLCluster manages the data, including backup files. Therefore we allow a user to set the backup policy in the MySQLCluster on their own. And we launch the backup jobs in their namespace.
Allowing another user to control backup data is contrary to this concept. So we concluded if the cluster administrator wanted to manage backup files, we needed another mechanism from the current BackupPolicy.

However, this usage is different from my team. So we have decided not to implement it actively.
